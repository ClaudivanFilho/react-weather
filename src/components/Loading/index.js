import React, { Component } from 'react';
import PropTypes from 'prop-types';

import LoadingIcon from '../../images/sun_loader.gif'

import './index.scss';

export default class Loading extends Component {
  
  constructor(props) {
    super(props);
  }

  render() {
    return (
      this.props.active &&
      <div className="col-sm-12 text-center loading">
        <img src={LoadingIcon} width={100}/>
      </div>
    );
  }
}

Loading.propTypes = {
  active: PropTypes.bool
}


