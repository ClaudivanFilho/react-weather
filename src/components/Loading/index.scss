.loading {
  position: absolute;
  left: 0%;
  top: 70%;
  z-index: 99999;
}

// ======== Loading Transition CSS =====
.loading-enter {
  opacity: 0.01;
}

.loading-enter.loading-enter-active {
  opacity: 1;
  transition: opacity 500ms ;
}

.loading-leave {
  opacity: 1;
}

.loading-leave.loading-leave-active {
  opacity: 0.01;
  transition: opacity 20ms ease-out;
}
// ======= End Loading Transition ======
