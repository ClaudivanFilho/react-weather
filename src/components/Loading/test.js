import React from 'react';
import {shallow} from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import renderer from 'react-test-renderer';

import Loading from './index';

configure({ adapter: new Adapter() });

test('Test Loading Component - With active false', () => {
  const component = shallow(<Loading active={false}/>);
  expect(component.find('img').length).toEqual(0);
});

test('Test Loading Component - With active true', () => {
  const component = shallow(<Loading active={true}/>);
  expect(component.find('img').length).toEqual(1);
});

test('Test Loading Snapshot', () => {
  const component = renderer.create(
    <Loading active={true}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})