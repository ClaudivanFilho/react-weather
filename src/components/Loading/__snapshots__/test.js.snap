// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Loading Snapshot 1`] = `
<div
  className="col-sm-12 text-center loading"
>
  <img
    src="test-file-stub"
    width={100}
  />
</div>
`;
